doc/decisions/1_in_discussion/reference_counting.md

```diff
@@ -0,0 +1,286 @@
+# Reference Counting
+
+## Problem
+
+Elektra's data structures are shared between multiple owners.
+A `Key` can be owned by many different `KeySet`s.
+Both `Key` and `KeySet` can be used in bindings.
+If the languages of those bindings have automatic memory management, this also implies a shared ownership.
+
+To solve this issue of shared ownership, we need reference counting.
+
+An adjacent problem is that the names of `Key`s must be read-only while the `Key` is part of a `KeySet`.
+This can also be solved by a secondary counter just for `KeySet`s.
+
+There has also been [discussion](copy_on_write.md) about adding further data structures, which would need reference counters as well.
```

Comment:
```suggestion
There has also been [discussion](../1_in_discussion/copy_on_write.md) about adding further data structures, which would need reference counters as well.
```

Can we refer to other decisions with the directory they are in? This makes it **much** easier to move individual decisions, as updating links is just a single find/replace for `old_folder/decision_name.md` with `new_folder/decision_name.md".

Reply:
Good point, @markus2330 should add that to #4639 as well: "Links to decision should always contain the directory of the decision, even if the link is from another decision in the same directory."

Reply:
Good idea, I added d94ff201d0d6826e4a3d277fcd26c21ad8ecb792